tokio/src/sync/semaphore_ll.rs

```diff
@@ -185,6 +185,9 @@ impl Semaphore {
         num_permits: u16,
         permit: &mut Permit,
     ) -> Poll<Result<(), AcquireError>> {
+        // Keep track of task budget
+        ready!(crate::coop::poll_proceed(cx));
```

Comment:
This cannot go here. This `poll_acquire` fn is internal and the callers expect that a `Pending` return means something specific to the state of the semaphore. This call will need to go in the public fns.